packages/gatsby/src/utils/start-server.ts

```diff
@@ -469,6 +469,12 @@ module.exports = {
   }
 
   app.use(async (req, res) => {
+    // in this catch-all block we don't support POST so we should 404
+    if (req.method === `POST`) {
+      res.status(404).end()
```

Comment:
To be an HTTP pedant again (and I'm sure it makes no difference) but this should be `405`! 😄

Reply:
I don't know, this sounds like a 404. For me 405 (and I could be wrong) is when we don't support POST at all?

Reply:
Maybe. It's "Method Not Allowed", but I can see that the description could be read either way. In any case, not a big deal

Reply:
I checked what GitHub does for unspported methods and they also return 404 status (i.e. https://api.github.com/repos/gatsbyjs/gatsby/commits/1e702ae64e08e8b5d22ca448d870c9bb24954bf8 with GET is response, but with PATCH is 404)

Reply:
so 404 is fine IMO